project/buildSrc/src/main/kotlin/TokenGenerator.kt

```diff
@@ -0,0 +1,48 @@
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.util.*
+
+class TokenGenerator {
+    companion object {
+        private const val STRING_SIZE = 256
+        private const val STRING_LENGTH = STRING_SIZE / 8
+
+        private const val HASH_ALGORITHM = "SHA-256"
+        private const val TOKEN_DURATION : Long = 1000 * 60 * 60 * 24 * 20 //Time in milliseconds before token expiring
+
+        private val charPool = ('a'..'z') + ('A'..'Z') + ('0'..'9') //List of chars used in the random string generation
```

Comment:
Generate a random 256/8 byte array, and then convert it to a base64url (notice the url) string.